src/com/github/alex_moon/porcupi/contexts/Context.java

```diff
@@ -0,0 +1,59 @@
+package com.github.alex_moon.porcupi.contexts;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.github.alex_moon.porcupi.handlers.Handler;
+import com.github.alex_moon.porcupi.messages.Message;
+import com.github.alex_moon.porcupi.messages.PokeMessage;
+
+public abstract class Context {
+    protected long tid;
+    protected Handler handler;
+    protected Object monitor;
+    protected List<String> messages = new ArrayList<String>();
```

Comment:
Not quite. `String` is still the best thing here but it shouldn't really be "`message`"